src/frontend/src/pages/ChangeRequestDetailPage/ChangeRequestDetailsView.tsx

```diff
@@ -93,101 +100,136 @@ const ChangeRequestDetailsView: React.FC<ChangeRequestDetailsProps> = ({
-        open={dropdownOpen}
-        anchorEl={anchorEl}
-        onClose={handleDropdownClose}
-        anchorOrigin={{
-          vertical: 'bottom',
-          horizontal: 'right'
-        }}
-        transformOrigin={{
-          vertical: 'top',
-          horizontal: 'right'
-        }}
-      >
-        <MenuItem
-          to={`${routes.PROJECTS_NEW}?crId=${changeRequest.crId}&wbs=${projectWbsPipe(changeRequest.wbsNum)}`}
-          component={RouterLink}
-          onClick={handleDropdownClose}
-          disabled={!isUserAllowedToImplement}
-        >
-          <ListItemIcon>
-            <CreateNewFolderIcon fontSize="small" />
-          </ListItemIcon>
-          Create New Project
-        </MenuItem>
-        <MenuItem
-          component={RouterLink}
-          to={`${routes.WORK_PACKAGE_NEW}?crId=${changeRequest.crId}&wbs=${projectWbsPipe(changeRequest.wbsNum)}`}
-          disabled={!isUserAllowedToImplement}
-          onClick={handleDropdownClose}
-        >
-          <ListItemIcon>
-            <PostAddIcon fontSize="small" />
-          </ListItemIcon>
-          Create New Work Package
-        </MenuItem>
-        <MenuItem
-          component={RouterLink}
-          to={`${routes.PROJECTS}/${wbsPipe(changeRequest.wbsNum)}?crId=${changeRequest.crId}&edit=${true}`}
-          disabled={!isUserAllowedToImplement}
-          onClick={handleDropdownClose}
-        >
-          <ListItemIcon>
-            <EditIcon fontSize="small" />
-          </ListItemIcon>
-          Edit {changeRequest.wbsNum.workPackageNumber === 0 ? 'Project' : 'Work Package'}
-        </MenuItem>
-      </Menu>
-    </div>
+    <ActionsMenu
+      buttons={[
+        {
+          title: 'Create New Project',
+          onClick: () =>
+            history.push(`${routes.PROJECTS_NEW}?crId=${changeRequest.crId}&wbs=${projectWbsPipe(changeRequest.wbsNum)}`),
+          disabled: !isUserAllowedToImplement,
+          icon: <CreateNewFolderIcon fontSize="small" />
+        },
+        {
+          title: 'Create New Work Package',
+          onClick: () =>
+            history.push(
+              `${routes.WORK_PACKAGE_NEW}?crId=${changeRequest.crId}&wbs=${projectWbsPipe(changeRequest.wbsNum)}`
+            ),
+          disabled: !isUserAllowedToImplement,
+          icon: <PostAddIcon fontSize="small" />
+        },
+        {
+          title: `Edit ${changeRequest.wbsNum.workPackageNumber === 0 ? 'Project' : 'Work Package'}`,
+          onClick: () =>
+            history.push(`${routes.PROJECTS}/${wbsPipe(changeRequest.wbsNum)}?crId=${changeRequest.crId}&edit=${true}`),
+          disabled: !isUserAllowedToImplement,
+          icon: <EditIcon fontSize="small" />
+        }
+      ]}
+      title="Implement Change Request"
+    />
   );
 
-  const actionDropdown = changeRequest.accepted ? implementCrDropdown : unreviewedActionsDropdown;
+  const actionDropdown = changeRequest.accepted ? implementCrDropdown : renderUnreviewedActionsDropdown;
```

Comment:
ok now that I'm reading through all this again, all this dropdown is distinct and complex enough to just be in its own file. Can you move all the code that's being used to get actionDropdown here into another file called ChangeRequestActionMenu.tsx and just make it all a new component there

Reply:
I can give some more comments about how to organize the code/ remove unnecessary code once that's done